### DISCUSSION TITLE
Jargon

### DISCUSSION THREAD

**Discussion by meganindya** [sugarlabs/musicblocks-v4#10] [Ideas]
We need to settle on a set of jargon, which we shall use throughout the development and maintenance cycles, for components, units, data structures, etc. We're often at a dilemma in the current _Music Blocks_.

- Entry point of the application: ________ (analogous to `Activity` in _Music Blocks_).
- Component that handles program execution: ________ (analogous to `Logo` in _Music Blocks_).
- Data Structure to store the program's instruction sequence: ________ (analogous to `blockList` and `macros` in _Music Blocks_).
- UI bars that contain the buttons: ________ (analogous to `Toolbar` in _Music Blocks_).
- The Sprite: ________ (analogous to `Turtle` in _Music Blocks_).
- (if required) Subcomponent of the Sprite that handles "music": ________ (analogous to `Singer` in _Music Blocks_).
- (if required) Subcomponent of the Sprite that handles "art": ________ (analogous to `Painter` in _Music Blocks_).

<br />

With respect to the Model-View-Architecture

- `Engine` is the `Model`.
- `Client` is the `View`.
- `Broker` is the `Controller`.

@walterbender @pikurasa

**Comment by walterbender** [comment]
+1

**Comment by pikurasa** [comment]
Sounds good to me.
